infra/deploy.sh

```diff
@@ -42,6 +42,8 @@ terraform {
     bucket = "fp-${fp_context}-terraform-state"
     key    = "infrastructure.tfstate"
   }
+
+  required_version = "~> 0.12"
```

Comment:
```suggestion
  required_version = "~> 0.12.29"
```

I didn't get the appropriate error (running 0.13.0) without including the patch.

Everything else look good!

From the docs:

> ~>: Allows the specified version, plus newer versions that only increase the most specific segment of the
> specified version number. For example, ~> 0.9 is equivalent to >= 0.9, < 1.0, and ~> 0.8.4, is equivalent to >= 
> 0.8.4, < 0.9. This is usually called the pessimistic constraint operator.